src/core/process_management/gt_taskgroup.cpp

```diff
@@ -122,12 +123,28 @@ GtTaskGroup::read(const QString& projectPath,
         auto newTask = m_pimpl->createTaskFromFile(
                     dir.absoluteFilePath(e.toString() + S_TASK_FILE_EXT));
 
-        if (newTask)
+        if (!newTask)
         {
+            gtError() << QObject::tr("Invalid task file (%1)").arg(e.toString());
+            continue;
+        }
+
+        if (newTask.get()->isDummy())
+        {
+            gtDebug().medium().nospace() << "dummy task created ("
+                                         << newTask->uuid() << ")";
+        }
+        else
+        {
+            GtTask* gtTask = dynamic_cast<GtTask*>(newTask.get());
+            assert(gtTask != nullptr && "newTask must be of type GtTask.");
```

Comment:
```suggestion
            assert(qobject_cast<GtTask*>(newTask.get()) && "newTask must be of type GtTask.");
```

This way, the cast is only performed in debug mode.